### PR TITLE
feat(page): supports the `enter` key for image caption

### PR DESCRIPTION
**Pull request by zqran**
close: #4904

https://github.com/toeverything/blocksuite/assets/15389209/4cf6e1f5-eef2-4f42-aa0d-e93a1bc1a709

### CODE REVIEW

File: packages/blocks/src/image-block/image-block.ts [hunk 1]
```diff
@@ -411,6 +413,37 @@ export class ImageBlockComponent extends BlockElement<ImageBlockModel> {
     };
   }
 
+  private _onCaptionKeydown(e: KeyboardEvent) {
+    if (e.key === 'Enter') {
```
Comment (Flrande) [inline packages/blocks/src/image-block/image-block.ts:417]: Need to check `e.isComposing`

File: packages/blocks/src/image-block/image-block.ts [hunk 2]
```diff
@@ -411,6 +413,37 @@ export class ImageBlockComponent extends BlockElement<ImageBlockModel> {
...
+      if (!parent) return;
+      const index = parent.children.indexOf(model);
+      const id = page.addBlock(
+        'affine:paragraph',
+        { text: new Text(nextBlockText) },
+        parent,
+        index + 1
+      );
+      asyncFocusRichText(model.page, id, {
+        index: nextBlockText.length,
```
Comment (Flrande) [inline packages/blocks/src/image-block/image-block.ts:441]: I think it will be more reasonable to use `0` to keep index when we split the text.

**Comment by Flrande** [comment]
Can you add a test?

File: packages/blocks/src/image-block/image-block.ts (hunk 1 above)
Comment (zqran) [reply packages/blocks/src/image-block/image-block.ts:418]: done

**Pull request merged**